routes/pangolin.js

```diff
@@ -72,30 +69,43 @@ async function calcAddresses() {
   return num_addresses;
 }
 
-// WIP not correct, need to get number of swaps, not number of txs
-async function calcAvg() {
-  let result = await client.request(factoryQuery)
-  let totalVolumeETH = result['pangolinFactories'][0]['totalVolumeETH']
-  let txCount = result['pangolinFactories'][0]['txCount']
-
-  let avg = totalVolumeETH / txCount
-
-  let avaxPrice = await getAvaxPrice()
+/**
+ * Helper to discover number of swap transactions
+ */
+async function getSwapsNumber(i = 0, j = 100000) {
+  let { swaps } = await client.request(swapQuery, { skip: j });
+  let is_swaps = (swaps.length) ? swaps.length : 0;
+
+  if((is_swaps < 100 && is_swaps > 0) || j === 0) {
+    return j + is_swaps;
+  } else {
+    return is_swaps ? await getSwapsNumber(j, (j + ((j-i) * 2))) : await getSwapsNumber(i, Math.floor((j - ((j-i) / 2))));
+  }
+}
 
+/**
+ * Calculate average swap transactions on USD
+ */
+async function calcAvg() {
+  let result = await client.request(factoryQuery);
+  let totalVolumeETH = result['pangolinFactories'][0]['totalVolumeETH'];
+  let avaxPrice = await getAvaxPrice();
+  let swapCount = await getSwapsNumber();
+  let avg = totalVolumeETH / swapCount;
   let avgUSD = avg * avaxPrice;
 
   return avgUSD;
 }
 
-// WIP not correct, also need swap count, not tx count
+/**
+ * Calculate median on swap transactions USD
+ */
 async function calcMedian() {
+  let swapCount = await getSwapsNumber();
+  let medianIndex = Math.floor(swapCount / 2);
+  let median = await client.request(swapQuery, { first: 1, skip: medianIndex, orderBy: 'amountUSD' });
 
-  let result = await client.request(factoryQuery)
-
-  let txCount = result['pangolinFactories'][0]['txCount']
-
-  let medianIndex = Math.round(txCount / 2)
-
+  return parseFloat(median['swaps'][0]['amountUSD']);
```

Comment:
This is going to be confusing, but this is actually in units of AVAX and needs to be multiplied by the price of AVAX. Otherwise looks good!

Reply:
This part is solve. I just added a comment about _totalVolumeETH._